paddle/fluid/framework/op_desc.cc

```diff
@@ -202,6 +202,57 @@ std::vector<std::string> OpDesc::AttrNames() const {
 }
 
 void OpDesc::SetAttr(const std::string &name, const Attribute &v) {
+  // NOTICE(minqiyang): pybind11 will take the empty list in python as
+  // the std::vector<int> type in C++; so we have to change the attr's type
+  // here if we meet this issue
+  proto::AttrType attr_type = static_cast<proto::AttrType>(v.which() - 1);
+  if (attr_type == proto::AttrType::INTS &&
+      boost::get<std::vector<int>>(v).size() == 0u) {
+    proto::OpProto proto = OpInfoMap::Instance().Get(Type()).Proto();
+    // Find current attr via attr name and set the correct attribute value
+    for (int i = 0; i != proto.attrs_size(); ++i) {
+      const proto::OpProto::Attr &attr = proto.attrs(i);
+      if (attr.name() == name) {
```

Comment:
Here, we can extract a method, like 
```cpp
const proto::OpProto::Attr& GetAttr(const proto::OpProto& proto, const std::string& name);
```

Reply:
Thanks, I will extract a method in next commit